Modix.Services/Giveaways/GiveawayService.cs

```diff
@@ -39,63 +36,57 @@ public GiveawayService(IAuthorizationService authorizationService)
         }
 
         /// <inheritdoc />
-        public async ValueTask<WinnersResult> GetWinnersAsync(IUserMessage message, int count)
+        public GiveawayResult GetWinners(IUserMessage message, int count)
         {
-            _authorizationService.RequireClaims(AuthorizationClaim.HostGiveaway);
+            _authorizationService.RequireClaims(AuthorizationClaim.ExecuteGiveaway);
 
             if (count <= 0)
             {
-                return new WinnersResult("You need to request at least one winner.");
+                return GiveawayResult.FromError("You need to request at least one winner.");
             }
 
             if (count > MaximumWinners)
             {
-                return new WinnersResult($"You can only have a maximum of {MaximumWinners} winners per giveaway.");
+                return GiveawayResult.FromError($"You can only have a maximum of {MaximumWinners} winners per giveaway.");
             }
 
-            var reactors = await GetReactorsAsync(message);
+            var reactors = GetReactors(message);
 
             if (reactors.Length == 0)
             {
-                return new WinnersResult("Cannot choose any winners, because nobody entered the giveaway.");
+                return GiveawayResult.FromError("Cannot choose any winners, because nobody entered the giveaway.");
             }
             else if (reactors.Length <= count)
             {
                 var winners = reactors.Select(x => x.Id)
                     .ToImmutableArray();
 
-                return new WinnersResult(winners);
+                return GiveawayResult.FromSuccess(winners);
             }
             else
             {
                 var winners = DetermineWinners(reactors, count);
 
-                return new WinnersResult(winners);
+                return GiveawayResult.FromSuccess(winners);
             }
         }
 
-        private async Task<ImmutableArray<IUser>> GetReactorsAsync(IUserMessage message)
-        {
-            var reactors = await message.GetReactionUsersAsync(_giveawayEmoji, int.MaxValue)
-                .FlattenAsync();
-
-            return reactors.Where(x => !x.IsBot)
+        private ImmutableArray<IUser> GetReactors(IUserMessage message)
+            => message.GetReactionUsersAsync(_giveawayEmoji, int.MaxValue)
+                .ToEnumerable()
```

Comment:
Uhhhh....what? How are we calling an `Async` method and somehow not end up returning a `Task`? Does this even compile?

Reply:
Rx is weird, man. It does compile. As far as I can figure, it's either something like this or the `FlattenAsync` thing I had before